lightning/src/ln/channel.rs

```diff
@@ -7123,7 +7123,9 @@ where
 					log_trace!(logger, " ...removing outbound AwaitingRemovedRemoteRevoke {}", &htlc.payment_hash);
 					if let OutboundHTLCOutcome::Failure(mut reason) = outcome.clone() { // We really want take() here, but, again, non-mut ref :(
 						if let (Some(timestamp), Some(now)) = (htlc.send_timestamp, now) {
-							let hold_time = u32::try_from(now.saturating_sub(timestamp).as_millis()).unwrap_or(u32::MAX);
+							const HOLD_TIME_GRANULARITY_MS: u128 = 100;
+
+							let hold_time = u32::try_from(now.saturating_sub(timestamp).as_millis() / HOLD_TIME_GRANULARITY_MS).unwrap_or(u32::MAX);
```

Comment:
No test coverage on this. Not sure what the best way is to test something involving time without an injectable time provider.

Reply:
It did feel a bit bare still. I added an `onion_route_tests` test for it.